### PR TITLE
Add abstract object changes

### PR DESCRIPTION
**Pull request by SupunS**

### CODE REVIEW

File: examples/abstract-objects/abstract-objects.bal [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import ballerina/io;
+
+// Defines an abstract object called 'Person'. It describes the type of
+// each field and method. However it cannot have a constructor method,
+// or any attached functions.
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.bal]: Minor changes:
"// Define an abstract object called 'Person'. Be sure to describe
// each field type and method. Note that an abstract object cannot have a constructor method
// or any attached functions."

File: examples/abstract-objects/abstract-objects.bal [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    // Function declarations can be within the object. But the function cannot 
+    // have a body.
+    function getFullName() returns string;
+
+    function checkAndModifyAge(int condition, int a);
+};
+
+// A non-abstract object called 'Employee', which is structurally equivalent
+// to 'Person'. It cannot have any member functions without a body.
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.bal]: Minor changes:
"// Define a non-abstract object called 'Employee', which is structurally equivalent
// to 'Person'. Note that a non-abstract object cannot have any member functions without a body."

File: examples/abstract-objects/abstract-objects.bal [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    // Member function should have a body
+    function getFullName() returns string {
+        return firstName + " " + lastName; 
+    }
+
+    //Otherwise must be defined outside.
+    function checkAndModifyAge(int condition, int a);
+};
+
+// Implementation for the declared function.
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.bal]: Minor change:
"// Implement the declared function."

File: examples/abstract-objects/abstract-objects.description [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// In Ballerina, objects can be abstract. These objects only describes the type of
+// each field and method. These objects cannot be initialized, and does not have a
+// default value.
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.description]: Minor changes:
"// In Ballerina, objects can be abstract. Abstract objects should only describe
// each field type and method. An abstract object cannot be initialized, and does not have a
// default value."

File: examples/abstract-objects/abstract-objects.bal [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        self.age = a;
+    }
+}
+
+public function main() {
+    // An abstract object type cannot be initialized. It does not have 
+    // an implicit initial value.
+
+    // A value using the non-abstract object 'Employee' can be initialized. 
+    // It can be then assigned to the abstract object type variable.
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.bal]: Can we say:
"// Initialize a value using the non-abstract object 'Employee',  
 // and then assign the value to the abstract object type variable."

File: examples/abstract-objects/abstract-objects.description [hunk 6]
```diff
@@ -1,3 +1,3 @@
-// In Ballerina, objects can be abstract. These objects only describes the type of
-// each field and method. These objects cannot be initialized, and does not have a
-// default value.
+// In Ballerina, objects can be abstract. Abstract objects only describes the type of
+// each field and method. An abstract object cannot be initialized, and does not have
```
Comment (Nashaath) [inline examples/abstract-objects/abstract-objects.description]: Minor correction:
"// In Ballerina, objects can be abstract. Abstract objects only describe 
// each field type and method. An abstract object cannot be initialized, and does not have"

File: examples.txt [hunk 7]
```diff
@@ -157,6 +157,7 @@ Objects
 Object Initializer
 Object Member Functions
 Object Assignability
+Abstract Objects
```
Comment (Shan1024) [inline examples.txt]: We need to add the new example to travis config as well.

Comment (SupunS) [reply examples.txt]: updated

**Review by Shan1024 — APPROVED** [review]

File: examples/abstract-objects/abstract-objects.bal [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+type Employee object {
+    public int age;
+    public string firstName;
+    public string lastName;
+
+    // Non-abstract object can have a constructor method.
+    new(age, firstName, lastName) {
+    }
+
+    // Member function should have a body
```
Comment (anupama-pathirage) [inline examples/abstract-objects/abstract-objects.bal]: Shall we add a period at the end

**Pull request merged**